src/ICal/ICal.php

```diff
@@ -1169,22 +1168,22 @@ protected function processRecurrences()
                 $initialStart             = new \DateTime($anEvent['DTSTART_array'][1]);
                 $initialStartTimeZoneName = $initialStart->getTimezone()->getName();
 
+                $endTimestamp = null;
                 if (isset($anEvent['DTEND'])) {
                     $initialEnd             = new \DateTime($anEvent['DTEND_array'][1]);
                     $initialEndTimeZoneName = $initialEnd->getTimezone()->getName();
+                    $endTimestamp = $initialEnd->getTimestamp();
                 } else {
                     $initialEndTimeZoneName = $initialStartTimeZoneName;
                 }
 
                 // Get Start timestamp
                 $startTimestamp = $initialStart->getTimestamp();
 
-                if (isset($anEvent['DTEND'])) {
-                    $endTimestamp = $initialEnd->getTimestamp();
-                } elseif (isset($anEvent['DURATION'])) {
+                if (isset($anEvent['DURATION'])) {
                     $duration = end($anEvent['DURATION_array']);
                     $endTimestamp = $this->parseDuration($anEvent['DTSTART'], $duration);
-                } else {
+                } elseif(!isset($anEvent['DTEND'])) {
                     $endTimestamp = $startTimestamp;
                 }
```

Comment:
Why the swap?

Reply:
What do you mean by "swap"? The variable assignment or logic?  $endTimestamp is already set in line 1175 if the event specifies a DTEND. This mirrors that logic above to set $endTimestamp if there is no DTEND specified and there is no DURATION.